reference/functions/pgroonga-condition.md

```diff
@@ -0,0 +1,160 @@
+---
+title: pgroonga_condition function
+upper_level: ../
+---
+
+# `pgroonga_condition()` function
+
+## Summary
+
+`pgroonga_condition()` function return `pgroonga_condition` type value.
+The function and the type have same name, but they are two different things.
+`pgroonga_condition` type represent complicated conditional expressions, such as `pgroonga_full_text_search_condition` type and `pgroonga_full_text_search_condition_with_scorers` type. 
+
+`pgroonga_condition()` function is a useful function to make the `pgroonga_condition` type value.
+It allows to make the `pgroonga_condition` type value by designating the specific attribute value.
+
+There were not this kind of useful functions for `pgroonga_full_text_search_condition` type and `pgroonga_full_text_search_condition_with_scorers` type, so designating all attribute values was necessary to make the value.
+
+Therefore, you need to designate `NULL` for disused attribute value as follows when `pgroonga_full_text_search_condition` type and `pgroonga_full_text_search_condition_with_scorers` type are used to avoid making all the values.
+
+```
+title &@~ ('keyword', NULL, 'index_name')::pgroonga_full_text_search_condition
+title &@~ ('keyword', ARRAY[1,1,1,5,0], NULL, 'index_name')::pgroonga_full_text_search_condition_with_scorers
+```
+
+It was not possible for existing value creation methods to make new attribute value while keeping backward compatibility.
+Thus, it was necessary to add a new type every time a new attribute value is added, such as `pgroonga_full_text_search_condition_with_XXX` type.
+For example, `pgroogna_full_text_search_condition_with_scorers` type was added because of the added new attribution.
+
+The difference between `pgroonga_full_text_search_condition` type and `pgroonga_full_text_search_condition_with_scorers` type is whether `scorers` exist or not. If `scorers` is added to `pgroonga_full_text_search_condition` type, every users are required to insert new 'NULL' to make `pgroonga_full_text_search_condition` type regardless of `scorers` usage.
+
+However, installing `pgroonga_condition()` function to make new `pgroonga_condition` type value let a new attribution to be added while keeping backward compatibility.
+It is because `pgroonga_condition()` function absorb incompatibility.
+
+`pgroonga_condition()` function let current writing style when a new attribute value is added because the function can leave out unnecessary attribution value as following sample.
```

Comment:
```suggestion
`pgroonga_condition()` function lets current writing style when a new attribute value is added because the function can leave out unnecessary attribute value as following sample.
```